includes/class-boldgrid-inspirations-onboarding-tasks.php

```diff
@@ -0,0 +1,258 @@
+<?php
+/**
+ * File: boldgrid-inspirations/includes/class-boldgrid-inspirations-onboarding-tasks.php
+ *
+ * @package Boldgrid_Inspirations_Onboarding_Tasks
+ * @copyright BoldGrid.com
+ * @version $Id$
+ * @author BoldGrid.com <wpb@boldgrid.com>
+ */
+
+/**
+ * BoldGrid Inspirations Onboarding Tasks class.
+ *
+ * This class is responsible for determining which tasks to add to
+ * the onboarding process. This class should only run once per
+ * deployment, and only run when the deployment process is completed.
+ *
+ * @since 2.8.0
+ */
+class Boldgrid_Inspirations_Onboarding_Tasks {
+	/**
+	 * The onboarding tasks option name.
+	 *
+	 * @since 2.8.0
+	 *
+	 * @var string
+	 */
+	var $option_name;
+
+	/**
+	 * Inspirations Configs
+	 *
+	 * @since 2.8.0
+	 *
+	 * @var array
+	 */
+	var $configs;
+
+	/**
+	 * Constructor.
+	 *
+	 * @since 2.8.0
+	 */
+
+	public function __construct( $configs ) {
+		$this->configs     = $configs;
+		$this->option_name = $this->configs['onboarding_tasks_option'];
+	}
+
+	/**
+	 * Create Tasks for the onboarding process.
+	 *
+	 * @since 2.8.0
+	 *
+	 * @param array $install_options The install options.
+	 */
+	public function create_tasks( $install_options ) {
+		$available_tasks = $this->configs['available_onboarding_tasks'];
+		$active_tasks    = array();
+		$complete_tasks  = 0;
+
+		foreach ( $available_tasks as $task ) {
+			// If no active callback is provided, the task is always active.
+			if ( ! isset( $task['active_callback'] ) ) {
+				$active_tasks[] = array(
+					'id'            => $task['id'],
+					'task_complete' => $task['task_complete'],
+				);
+				continue;
+			}
+
+			// If the active callback is a callable function, run it.
+			if ( is_callable( $task['active_callback'] ) && true === call_user_func( $task['active_callback'], $install_options ) ) {
+				$active_tasks[] = array(
+					'id'            => $task['id'],
+					'task_complete' => $task['task_complete'],
+				);
+				continue;
+			}
+
+			// If the active callback is a method of this class, run it.
+			if ( is_callable( array( $this, $task['active_callback'] ) ) && true === call_user_func( array( $this, $task['active_callback'] ), $install_options ) ) {
+				$active_tasks[] = array(
+					'id'            => $task['id'],
+					'task_complete' => $task['task_complete'],
+				);
+				continue;
+			}
+		}
+
+		foreach ( $active_tasks as $task ) {
+			if ( true === $task['task_complete'] ) {
+				$complete_tasks++;
+			}
+		}
+
+		// Save the active tasks.
+		update_option( $this->option_name, $active_tasks );
+		update_option( $this->configs['onboarding_progress_option'], $complete_tasks / count( $active_tasks ) );
+	}
+
+	/**
+	 * Get Task
+	 *
+	 * @since 2.8.0
+	 *
+	 * @param string  $task_id       The task ID.
+	 * @param boolean $task_complete Whether the task is complete.
+	 *
+	 * @return array The task config array.
+	 */
+	public function get_task( $task_id, $task_complete ) {
```

Comment:
It seems strange to have a `$task_complete` as a parameter.

For example, if I said `get_post( 123, 'publish' )` as another developer it would be hard to understand why I'm passing a post status to a get_post... It should get post `123` regardless of the status.

Reply:
The reason for this, is that the config array contains a key that indicates whether it should already be marked complete or not. So when using this method to get the task, we want to use the current competion status when getting the task